chrome/browser/ui/browser_navigator.cc

```diff
@@ -532,8 +533,8 @@ std::unique_ptr<content::WebContents> CreateTargetContents(
   }
 
 #if BUILDFLAG(ENABLE_SECURE_EMBED)
-  create_params.secure_embed_delegate =
-      params.browser->window()->GetSecureEmbedDelegate();
```

Comment:
One thing that's kinda rough I don't feel great about: CreateParams is copyable (...mostly for benefits of GuestView, it seems), so this isn't a unique_ptr like it ought to be..

Reply:
Oh, wow, thank you, this is nice. It fixes this, and also means there is no need for public SecureEmbedConnector::Create, and SetEmbeddedWebContents can also go away since both WebContents can be passed to the constructor.

Reply:
GuestView has a simpler lifetime model - the outer WebContents owns (and outlives) the inner WebContents. GuestContents decouples the lifetime of inner / outer WebContents, and I think we should keep that. This will be important for use case like tab dragging.